src/frontend/src/optimizer/rule/apply_filter.rs

```diff
@@ -65,7 +70,8 @@ impl Rule for ApplyFilterRule {
             input,
             join_type,
             new_on,
-            (0..correlated_indices_len).collect_vec(),
+            apply.correlated_id(),
+            correlated_indices,
```

Comment:
Another nontrivial change that does the opposite of `apply_scan`: indices are changed from enumerate indices to `correlated_indices`. Could you shed more light on this?

Reply:
Same reason above.